pydicom/dataset.py

```diff
@@ -734,40 +734,45 @@ def convert_pixel_data(self):
                             handlers_tried)
                 self._pixel_array = None
                 self._pixel_id = None
-                if last_exception:
-                    raise last_exception
-                else:
+                exception_messages = []
+                for x in handler_exceptions:
+                    logger.debug("image handler exception", exc_info=x)
+                    exception_messages.append(str(x))
+                if not handler_exceptions:
                     msg = ("No available image handler could "
                            "decode this transfer syntax {}".format(
                                self.file_meta.TransferSyntaxUID))
                     raise NotImplementedError(msg)
+                else:
+                    raise NotImplementedError("\n".join(exception_messages))
+
             # is this guaranteed to work if memory is re-used??
             self._pixel_id = id(self.PixelData)
-            
```

Comment:
In know this is still work in progress, but in case in helps, I can elaborate I bit what I had in mind.
Basically, I would handle the case where no handler was found like this (out of my head, not tested):
``` python
module_names = [h.module_name() for h in pydicom.config.image_handlers
                            if not h.is_available() and h.supports_transfer_syntax(self)]
if 'NumPy' in module_names:
    msg = 'You need NumPy installed...'
else:
    msg = 'You need '
    if not numpy_handler.is_available():
        msg = 'NumPy and '
    if len(module_names) > 1:
        msg += 'one of ' + ' or '.join(module_names)
    else:
        msg += module_names[0]
    msg += ' installed to...'
```
This will produce a single message with the possible options.
This would need an `is_available` property that basically returns `has_numpy`, `has_gdcm` etc, and `module_name` property that would return something like `NumPy` or `jpeg_ls (CharPyLS)` for each handler.

The `is_available` property could also be used to iterate over the handlers first, leaving exceptions only for the cases where something goes wrong in the handler itself.